libimage/search.go

```diff
@@ -348,6 +331,41 @@ func searchRepositoryTags(ctx context.Context, sys *types.SystemContext, registr
 	return paramsArr, nil
 }
 
+// getDockerAuthConfig extracts a docker auth config from the CopyOptions.  Returns
+// nil if no credentials are set.
+func (options *SearchOptions) getDockerAuthConfig() (*types.DockerAuthConfig, error) {
```

Comment:
This is a code clone of https://github.com/containers/common/blob/main/libimage/copier.go#L172.

It.s probably worth refactoring the two functions into one that would username, credetials and ID token as arguments?  The `getDockerAuthConfig()` methods could call this function.

Reply:
Yes please.